tests/checkpointing/test_model_checkpoint.py

```diff
@@ -500,20 +501,20 @@ def test_none_monitor_top_k(tmpdir):
 
 def test_none_monitor_save_last(tmpdir):
     """ Test that a warning appears for save_last=True with monitor=None. """
-    with pytest.warns(UserWarning, match=r'ModelCheckpoint\(save_last=True, monitor=None\) is a redundant.*'):
+    with pytest.warns(UserWarning, match=r'ModelCheckpoint.*is a redundant.*'):
         ModelCheckpoint(dirpath=tmpdir, save_last=True)
     # These should not fail
     ModelCheckpoint(dirpath=tmpdir, save_last=None)
     ModelCheckpoint(dirpath=tmpdir, save_last=False)
 
 
-def test_model_checkpoint_none_monitor(tmpdir):
+def test_model_checkpoint_save_last_none_monitor(tmpdir, caplog):
```

Comment:
isn't it safer to have separate test, so that we know they independently work?

Reply:
in this test here we also have `save_top_k=-1`, but according to the PR description, I think we need a test for
monitor=None, save_last=True and the test should show that two files are saved:
`epoch=X.ckpt` and `last.ckpt`, where X is the last epoch and they are two identical files.
is this correct?

Reply:
> so that we know they independently work?

Which things should be independent?

> in this test here we also have save_top_k=-1, but according to the PR description, I think we need a test for
monitor=None, save_last=True

Top k -1 or >0 does not matter.

> I think we need a test for
monitor=None, save_last=True and the test should show that two files are saved:
epoch=X.ckpt and last.ckpt, where X is the last epoch and they are two identical files.

That is the purpose of this (adapted) test, although it's not testing that the contents are exactly the same